cmd/cosign/cli/verify/verify_blob.go

```diff
@@ -253,33 +264,38 @@ func verifyRekorEntry(ctx context.Context, ko sign.KeyOpts, pubKey signature.Ver
 	if err != nil {
 		return err
 	}
-	var pubBytes []byte
-	if pubKey != nil {
-		pubBytes, err = sigs.PublicKeyPem(pubKey, signatureoptions.WithContext(ctx))
-		if err != nil {
-			return err
+	// We may already have the entry from search lookup if a key was not provided.
+	// If not, find the tlog entry by the proposed entry and retrieve the UUID and entry to verify.
+	if e == nil {
+		var pubBytes []byte
+		if pubKey != nil {
+			pubBytes, err = sigs.PublicKeyPem(pubKey, signatureoptions.WithContext(ctx))
+			if err != nil {
+				return err
+			}
 		}
-	}
-	if cert != nil {
-		pubBytes, err = cryptoutils.MarshalCertificateToPEM(cert)
+		if cert != nil {
+			pubBytes, err = cryptoutils.MarshalCertificateToPEM(cert)
+			if err != nil {
+				return err
+			}
+		}
+		uuid, e, _, err = cosign.FindTlogEntry(ctx, rekorClient, b64sig, blobBytes, pubBytes)
 		if err != nil {
 			return err
 		}
 	}
-	uuid, index, err := cosign.FindTlogEntry(ctx, rekorClient, b64sig, blobBytes, pubBytes)
+	verifiedEntry, err := cosign.VerifyTLogEntry(ctx, rekorClient, e, uuid)
```

Comment:
FYI it was very unclear that FindTlogEntry performed the merkle verification as well. I rescoped this so FindTlogEntry only performs the lookup by proposed entry (which also reduces the extra call to findtlogentry when we already found it in the keyless flow)